src/server/cmd/pachctl/shell/shell.go

```diff
@@ -93,6 +93,10 @@ func newShell(rootCmd *cobra.Command, maxCompletions int64) *shell {
 }
 
 func (s *shell) executor(in string) {
+	if in == "exit" {
+		os.Exit(1)
```

Comment:
```suggestion
		os.Exit(0)
```
I think this should be a non-error exit, that's what bash and zsh seem to do.